### PR TITLE
DEPR: change pd.concat sort=None to sort=False

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/reshape/concat.py [hunk 1]
```diff
@@ -82,18 +82,14 @@ def concat(
...
-
-        Explicitly pass ``sort=True`` to silence the warning and sort.
-        Explicitly pass ``sort=False`` to silence the warning and not sort.
-
+        is 'outer'.
         This has no effect when ``join='inner'``, which already preserves
         the order of the non-concatenation axis.
 
         .. versionadded:: 0.23.0
+        .. versionchanged:: 1.0.0
```
Comment (TomAugspurger) [inline pandas/core/reshape/concat.py]: Should note what changed here I think.

```suggestion
        .. versionchanged:: 1.0.0

           Changed to not sort by default.
```

File: pandas/tests/frame/test_join.py [hunk 2]
```diff
@@ -213,12 +213,6 @@ def test_suppress_future_warning_with_sort_kw(sort_kw):
...
         expected = expected.reindex(index=["c", "a", "b"])
 
-    if sort_kw is None:
-        # only warn if not explicitly specified
-        ctx = tm.assert_produces_warning(FutureWarning, check_stacklevel=False)
-    else:
-        ctx = tm.assert_produces_warning(None, check_stacklevel=False)
-
-    with ctx:
+    with tm.assert_produces_warning(None, check_stacklevel=False):
```
Comment (WillAyd) [inline pandas/tests/frame/test_join.py]: We fail on warnings now right (?). I think could probably get rid of this altogether

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/tests/frame/test_join.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_join.py]: do we fail on warnings for all builds, or just the one?

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**